### ISSUE TITLE
scheduler:max_cores does not override default value for nr_reduces in v0.2.4

### ISSUE THREAD

**Issue by davin** [discoproject/disco#90]
For the script below, when nr_reduces is set to 2, upon job completion, the status page shows Reduce/Done = 2.  But when scheduler{"max_cores":2} is set instead, the page shows Reduce/Done = 1.

---

<code>
from disco.core import Disco, result_iterator, Params
from disco.func import chain_reader, nop_reduce

def _mapDiag(e, params):
   if type(e) == tuple:
      e = e[0]
   output = []
   elems = e.split(";")
   for elem in elems:
      i, j, val = elem.split(",")
      if i == j:
         output.append(("%s,%s,%f" % (i,j,float(val)), ""))
   return output

disco = Disco("http://localhost:8989")
A = []
for i in range(0, 10):
        for j in range(0, 10):
         A.append("raw://%d,%d,%f" % (i,j, i*j))
myJob = disco.new_job(input=A, name="diag1", scheduler={"max_cores":2}, map=_mapDiag)
resDiag = myJob.wait(clean=False)
resAdd = disco.new_job(input=A, name="diag2", scheduler={"max_cores":2}, map=_mapDiag).wait(clean=False) 
resAdd = disco.new_job(input=resDiag+resAdd, name="foo", nr_reduces=2, map_reader=chain_reader, reduce=nop_reduce).wait(clean=False)
#resAdd = disco.new_job(input=resDiag+resAdd, name="foo", scheduler={"max_cores":2}, map_reader=chain_reader, reduce=nop_reduce).wait(clean=False)
</code>

**Comment by davin** [comment]
I believe I made an incorrect assumption that max_cores should automatically set nr_reduces to the same value.  My understanding now is that max_cores only controls the max # of concurrent reduces running at a time and has nothing to do with the total number of reduces produced.  So for example, nr_reduces=10 while max_cores=2 means at most only 2 reduces can be running side by side but in the end, you still have 10 reduces executed.  Also, by default, nr_reduces=1 which explains why I saw Reduce/Done = 1 as reported earlier.
